### PR TITLE
feat(ip.firewall): firewall create deny rule

### PR DESCRIPTION
**Pull request by Steffy29**
| Question         | Answer
| ---------------- | ---
| Branch?          | `develop`
| Bug fix?         | no
| New feature?     | yes
| Breaking change? | no
| Tickets          | Fix #UXCT-488
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [x] Only FR translations have been updated
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~Breaking change is mentioned in relevant commits~

## Description

- add link to create deny rule
- set default values for deny rule

## Related

- [ ] #9874

### CODE REVIEW

**Review by sachinrameshn — CHANGES_REQUESTED** [review]
Can you please change the commit message to include ticket # and remove other things

Ex:
feat(universe): title

ref: ticket#
description

Signed-off-by: Stephanie Moallic [stephanie.moallic@corp.ovh.com]

**Review by JacquesLarique — APPROVED** [review]

**Review by sachinrameshn — APPROVED** [review]

File: packages/manager/apps/dedicated/client/app/ip/components/list/firewall/ip-ip-firewall.controller.js [hunk 1]
```diff
@@ -127,13 +149,7 @@ export default /* @ngInject */ function IpFirewallCtrl(
...
 
   function changeRuleStatus(index, state) {
-    if (
-      index >= 0 &&
-      index < self.rules.list.results.length &&
-      self.rules.list.results[index]
-    ) {
-      self.rules.list.results[index].state = state;
-    }
+    self.rules.list.results[index].state = state;
```
Comment (anooparveti) [inline packages/manager/apps/dedicated/client/app/ip/components/list/firewall/ip-ip-firewall.controller.js:152]:
```suggestion
    self.rules.list.results[index]?.state = state;
```

File: packages/manager/apps/dedicated/client/app/ip/components/list/firewall/ip-ip-firewall.controller.js [hunk 2]
```diff
@@ -194,13 +202,10 @@ export default /* @ngInject */ function IpFirewallCtrl(
...
-            if (!isDenyRule) {
-              self.denyMessage = true;
-              // TODO : set the correct value
-              self.denyUrl = '#';
-            } else {
-              self.denyMessage = false;
-            }
+            const index = findIndex(self.rules.list.results, {
+              action: 'deny',
+            });
```
Comment (anooparveti) [inline packages/manager/apps/dedicated/client/app/ip/components/list/firewall/ip-ip-firewall.controller.js:207]: **suggestion:** Native methods on `Array.prototype` is the preferred

File: packages/manager/apps/dedicated/client/app/ip/components/list/firewall/ip-ip-firewall.controller.js [hunk 3]
```diff
@@ -442,4 +426,18 @@ export default /* @ngInject */ function IpFirewallCtrl(
...
   };
+
+  self.createDenyRule = () => {
+    self.rulesLoading = true;
+
+    // Initialize action value
+    self.rule.action = 'deny';
+
+    // Initialize rule sequence
+    const index = findIndex(self.constants.sequences, { name: 19 });
```
Comment (anooparveti) [inline packages/manager/apps/dedicated/client/app/ip/components/list/firewall/ip-ip-firewall.controller.js:437]: **suggestion:** Native methods on `Array.prototype` is the preferred

File: packages/manager/apps/dedicated/client/app/ip/components/list/firewall/ip-ip-firewall.html [hunk 4]
```diff
@@ -37,10 +37,14 @@ <h4 data-translate="ip_firewall_title"></h4>
...
-            data-translate="ip_firewall_deny"
-            data-translate-values="{ 'denyUrl': IpFirewallCtrl.denyUrl}"
-        ></span>
+        <span data-translate="ip_firewall_deny_part_1"></span>
+        <a
+            href="javascript:void(0)"
+            data-ng-click="IpFirewallCtrl.createDenyRule()"
+        >
+            <span data-translate="ip_firewall_deny_link"></span>
+        </a>
```
Comment (anooparveti) [inline packages/manager/apps/dedicated/client/app/ip/components/list/firewall/ip-ip-firewall.html:46]:
```suggestion
        <a
            href="#"
            data-ng-click="IpFirewallCtrl.createDenyRule()"
        >
            <span data-translate="ip_firewall_deny_link"></span>
        </a>
```

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/apps/dedicated/client/app/ip/components/list/firewall/ip-ip-firewall.html (hunk 4 above)
Comment (Steffy29) [reply packages/manager/apps/dedicated/client/app/ip/components/list/firewall/ip-ip-firewall.html:46]: Nope, if I do your suggestion, the page will be reloaded and it's not what I want to do. I just want to have a link which  send to an action "createDenyRule".

**Pull request merged**